### ISSUE TITLE
PostgreSQL: 'date' not parsed 

### ISSUE THREAD

**Issue by jdsada30** [sqlfluff/sqlfluff#1091]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

## Expected Behaviour
I would expect statements with keyword 'date' to be parsed by sqlfluff, as they are supported by PostgreSQL. 
## Observed Behaviour
>== [test3.sql] FAIL
L:   1 | P:  13 |  PRS | Found unparsable section: " '2001-02-16', DATE
                       | '2001-12-21'"
L:   1 | P:  47 | L012 | Implicit aliasing of column not allowed. Use explicit
                       | `AS` clause.
L:   1 | P:  47 | L029 | Keywords should not be used as identifiers.
L:   1 | P:  55 |  PRS | Found unparsable section: " (DATE '2001-10-30', DATE
                       | '2002-10-30')"
## Steps to Reproduce
Create a sql file with the following query:
> SELECT (DATE '2001-02-16', DATE '2001-12-21') OVERLAPS (DATE '2001-10-30', DATE '2002-10-30');
lint the file with the following command(s): 
>sqlfluff lint test.sql --dialect postgres or sqlfluff lint test.sql

## Version
I am using sqlfluff version 0.5.6 and python version 3.8.5.

This example comes from the official PostgreSQL [documentation](https://www.postgresql.org/docs/10/functions-datetime.html). I found similar results when executing queries from Github Project [verdict-project.](https://github.com/verdict-project)

**Comment by pwildenhain** [comment]
Doesn't seem like the problem is with `DATE` per se

`test_date.sql`
```sql
SELECT DATE '2001-02-16'
```

```
$ sqlfluff parse test_date.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:
     'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:
     ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    column_reference:
[L:  1, P:  8]      |                        identifier:
     'DATE'
[L:  1, P: 12]      |                    whitespace:
     ' '
[L:  1, P: 13]      |                    alias_expression:
[L:  1, P: 13]      |                        literal:
     "'2001-02-16'"
[L:  1, P: 25]      |            [META] dedent:
```

Seems like what is really confusing it is the `(dates) OVERLAPS (dates)` syntax
